Lib/inspect.py

```diff
@@ -410,12 +410,20 @@ def markcoroutinefunction(func):
     func._is_coroutine_marker = _is_coroutine_marker
     return func
 
+_agen_methods = ("__anext__", "aclose", "asend", "athrow")
+
 def iscoroutinefunction(obj):
     """Return true if the object is a coroutine function.
 
     Coroutine functions are normally defined with "async def" syntax, but may
     be marked via markcoroutinefunction.
     """
+    #check if obj is async gen method and returns awaitable
```

Comment:
Please, at least add tests for this feature.

Reply:
as soon as I fully understand the original question